vernac/vernacinterp.ml

```diff
@@ -111,6 +111,13 @@ let with_fail ~st f =
     if not !Flags.quiet || !test_mode
     then Feedback.msg_notice Pp.(str "The command has indeed failed with message:" ++ fnl () ++ msg)
 
+let with_succeed ~st f =
+  let () = ignore (f ()) in
+  Vernacstate.invalidate_cache ();
+  Vernacstate.unfreeze_interp_state st;
+  if not !Flags.quiet
+  then Feedback.msg_notice Pp.(str "The command succeeded and its effects have been reverted.")
```

Comment:
```suggestion
  then Feedback.msg_notice Pp.(str "The command has succeeded and it's effects have been reverted.")
```

Reply:
`its` is correct though

Reply:
Sorry, my bad.

Reply:
@SkySkimmer I still believe putting `has` is more clear, we can see what @jfehrle thinks too.

Reply:
It reads better with "has".

The past perfect tense (with "has") is used, among other things, to refer to events that "began in the past and _continued to the present time_", whereas the simple past tense is used to refer to "things that happened or existed _before now_".  The former seems more appropriate since the success occurs at the moment the message appears.  It's also the tense we used in the messages for `Fail`. See https://www.grammarly.com/blog/present-perfect-tense/ and https://www.grammarly.com/blog/simple-past/.

(That it reads better with "has" is an instantaneous judgment, then I crafted the rationale.)